lib/console.js

```diff
@@ -49,9 +55,32 @@ Console.prototype.error = Console.prototype.warn;
 
 
 Console.prototype.dir = function(object, options) {
-  this._stdout.write(util.inspect(object, util._extend({
-    customInspect: false
-  }, options)) + '\n');
+  this._stdout.write((this._group ? util.inspect(object, util._extend({
+        customInspect: false }, options)).replace(/^/gm, this._prefix) :
+          util.inspect(object, util._extend({ customInspect: false },
+              options)).replace(/^/gm, this._prefix)) + '\n');
+};
+
+
+Console.prototype.group = function() {
+  if (arguments.length) {
+    this.log.apply(this, arguments);
+  }
+  this._group++;
+  this._prefix = '  '.repeat(this._group);
+};
+
+
+Console.prototype.groupCollapsed = function() {
+  this.group.apply(this, arguments);
+};
+
+
+Console.prototype.groupEnd = function() {
+  if ( this._group > 0 ) {
+    this._group--;
+    this._prefix = (this._group ? '  '.repeat(this._group) : '');
```

Comment:
> No I only say, that your ( a ? b : c ) notation has unreachable part.

Actually, It does not have unreachable part, but I do see an extra `> 0` comparison that could be left out completely for optimization. Can you please define what is the unreachable part you see.

`(this._group ? '  '.repeat(this._group) : '');` evaluates to `''` empty string const only if `_group` has just been decreased to zero. Otherwise it just regenerates the `_prefix` with new string. It does not use `slice`, but it matches the code `group` uses for generating the prefix. 

I'm just trying to avoid avoiding unnecessary string concatenations and multiple methods in functions to generate the prefix. I chose `String.prototype.repeat` instead of string concatenation and `String.prototype.slice`.